living_documentation_generator/model/consolidated_issue.py

```diff
@@ -161,3 +168,38 @@ def generate_page_filename(self) -> str:
             return f"{self.number}.md"
 
         return page_filename
+
+    def generate_directory_path(self, issue_table: str) -> str:
+        """
+        Generate a directory path based on enabled features.
+
+        @param issue_table: The consolidated issue summary table.
+        @return: The generated directory path.
+        """
+        output_path = ActionInputs.get_output_directory()
+
+        # If structured output is enabled, create a directory path based on the repository
+        if ActionInputs.get_is_structured_output_enabled() and self.repository_id:
+            organization_name, repository_name = self.repository_id.split("/")
+            output_path = os.path.join(output_path, organization_name, repository_name)
+
+        # If grouping by topics is enabled, create a directory path based on the issue topic
+        if ActionInputs.get_is_grouping_by_topics_enabled():
+            # Extract labels from the issue table
+            labels = re.findall(r"\| Labels \| (.*?) \|", issue_table)
```

Comment:
I would add check & log when more than one topic label will be detected.

Reply:
[b6088b4](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/b6088b4c6cdc910837caa5ce05ebd8a465a5d338) I also added a logic that handles more than one Topic within one isssue in commit: [ecaffe0](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/ecaffe065a29c8fa8b513a4a98a1007c88af45b8)